SOS/pull_request_delivery/SERVICE/sos/sos.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * sos.h
+ *
+ * Created: 5/29/2023 2:40:32 AM
+ *  Author: Amr
+ */ 
+
+
+#ifndef SOS_H_
+#define SOS_H_
+
+#include "../../MCAL/dio/dio.h"
+#include "../../MCAL/timers/timer.h"
+#include "sos_cfg.h"
+#include "../../MCAL/Interrupt/exi.h"
+#include "../../HAL/button/button.h"
+#include "../../HAL/led/led.h"
+
+typedef struct
+{
+	u16		u16_task_id;
+	u16		u16_priority;
+	u16		u16_periodicity;
+	void (*pfTask)(void);
+
+} str_sos_config_task_t;
+
+
+typedef enum{
+	 SOS_STATUS_SUCCESS,
+	 SOS_STATUS_INVALID_STATE,
+	 SOS_STATUS_INVALID_ARGS,
+	 SOS_STATUS_TASK_ADDED,
+	 SOS_STATUS_TASK_DELETED,
+	 SOS_STATUS_TASK_MODIFIED,
+}enu_sos_error_status_t ;
+
+
+
+/**	
+ *																												
+ * @brief                                               :   Function Used to Initialize sos module				
+ *																												
+ * @return      SOS_STATUS_SUCCESS                      :   In case of Successful Operation						
+ *              SOS_STATUS_INVALID_STATE				:   In case the SOS is already Initialized				
+ *																												
+ */
```

Comment:
There are no prototypes for these functions (sos_scheduler,sos_start_check)

Reply:
because they are local helper functions, so no need to extern them in the .h file